.github/actions/pnpm/action.yml

```diff
@@ -23,7 +23,7 @@ runs:
 
     - if: ${{ inputs.cpu }}
       shell: bash
-      run: pnpm config set supportedArchitectures.cpu "${CPU_ARCH}"
+      run: echo "\nsupportedArchitectures:\n  cpu:\n    - ${CPU_ARCH}" >> pnpm-workspace.yaml
```

Comment:
Windows' bash doesn't intepret `\n` the same as Linux causes you problem, right?

I would use multiple `echo`s in this case:

```yaml
      run: |
        echo >> pnpm-workspace.yaml
        echo 'supportedArchitectures:' >> pnpm-workspace.yaml
        echo "  cpu: { $CPU_ARCH }" >> pnpm-workspace.yaml
```

Alternatively, there's `echo -n` but I don't know how well it would work and I prefer the multiple `echo` anyway.